### PR TITLE
Query: Dispose inner data readers for split query

### PR DESCRIPTION
**Pull request by smitpatel**
Part of #21441

### CODE REVIEW

File: src/EFCore.Relational/Query/Internal/SplitQueryingEnumerable.cs [hunk 1]
```diff
@@ -300,17 +306,24 @@ private async Task<bool> InitializeReaderAsync(DbContext _, bool result, Cancell
                 return result;
             }
 
-            public ValueTask DisposeAsync()
+            public async ValueTask DisposeAsync()
```
Comment (roji) [inline src/EFCore.Relational/Query/Internal/SplitQueryingEnumerable.cs:309]: Any reason the sync and async versions differ? Would be better to align them...

Comment (smitpatel) [reply src/EFCore.Relational/Query/Internal/SplitQueryingEnumerable.cs:309]: how are they differing?

Comment (roji) [reply src/EFCore.Relational/Query/Internal/SplitQueryingEnumerable.cs:309]: In various ways... the sync version disposes the main reader right away, the async copies it to a local and disposes at the end, the variable names differ...

Comment (smitpatel) [reply src/EFCore.Relational/Query/Internal/SplitQueryingEnumerable.cs:309]: You made it sound like there is functional difference.

Comment (smitpatel) [reply src/EFCore.Relational/Query/Internal/SplitQueryingEnumerable.cs:309]: Updated.

**Pull request merged**